src/Chapter3.hs

```diff
@@ -480,6 +518,42 @@ After defining the city, implement the following functions:
    and at least 10 living people inside
 -}
 
+data City = City {
+  castle :: Castle,
+  monument :: Monument,
+  houses :: Houses
+}
+
+data Castle = NoCastle 
+              | JustCastle String 
+              | CastleWithWall String
+
+data Monument = Library | Church
+
+type Houses  = [HouseHold]
+
+data HouseHold = OnePersonHouseHold 
+                |TwoPeopleHouseHold
+                |ThreePeopleHouseHold
+                |FourPeopleHouseHold
+
+buildWalls :: City -> City
+buildWalls city = case castle city of
+                    JustCastle name -> if length (houses city) >= 10 
```

Comment:
According to the task, it's not the number of houses that should be greater than 10, but the number of people inside houses. So you need to write a function of type `HouseHold -> Int` to calculate the total number of people in the city and then compare it to 10